### ISSUE TITLE
There are some problems with TopKPooling

### ISSUE THREAD

**Issue by USTBSCCE1028** [xxlya/BrainGNN_Pytorch#14]
![image](https://github.com/user-attachments/assets/7f8c3941-5ace-4b52-b3f7-699bf049beec)

Thank you very much for sharing the code, but there are some questions for you to answer.
self.pool1.weight,self.pool2.weight, TopKPooling does not have an attribute of weight, it will report an error, what did you intend to write here?
I've currently changed this to perm or score. but for the ABIDE dataset the test is only 50% correct is this reasonable?

**Comment by USTBSCCE1028** [comment]
This is in braingnn.py, the forward function of the Network class

**Comment by Tian-cnbj** [comment]
Thank you to the author for sharing the code. I encountered the same issue. May I ask what the weight attribute should be here?

**Comment by nibabel** [comment]
I solved this issue as follows.

First, pool1.weight and pool2.weight are used to project node features onto a learnable vector w.

In the paper, it states:
"To avoid the problem of identifiability, we propose unit loss."
This means that w needs to be output from the model in order to apply the unit loss.

However, PyG's TopKPooling does not have a .weight attribute.
By examining the TopKPooling implementation, we can see that:

```python
from torch_geometric.nn.pool.select import SelectTopK
...
class TopKPooling(torch.nn.Module):
    def __init__(self, ...):
        ...
        self.select = SelectTopK(in_channels, ratio, min_score, nonlinearity)
        ...
    def forward(self, ...):
        ...
        select_out = self.select(attn, batch)
        ...
```

It uses SelectTopK.

Looking into the SelectTopK implementation, at line 111, we find:
```python
class SelectTopK(Select):
    def __init__(self, ...):
        ...
        self.weight = torch.nn.Parameter(torch.empty(1, in_channels))    #--------> line 111
        ...
```

This means that we need to retrieve weight from SelectTopK, not directly from TopKPooling.


✅ Solution
In braingnn.py, modify the following line:

```python
return x,self.pool1.weight,self.pool2.weight, torch.sigmoid(score1).view(x.size(0),-1), torch.sigmoid(score2).view(x.size(0),-1)
```
to:
```python
return x,self.pool1.select.weight,self.pool2.select.weight, torch.sigmoid(score1).view(x.size(0),-1), torch.sigmoid(score2).view(x.size(0),-1)
```

After making this change, the model should work correctly without issues.

**Comment by mwu99** [comment]
Thank you for sharing the solution. May I know the version of your torch_geometric? I am not encountering this issue, but my test is also only about 50%, and I wonder how to fix it

**Comment by Tian-cnbj** [comment]
I have also conducted tests, and the accuracy on the ABIDE dataset is around 50%. This situation does exist in my case.

**Comment by nibabel** [comment]
> Thank you for sharing the solution. May I know the version of your torch_geometric? I am not encountering this issue, but my test is also only about 50%, and I wonder how to fix it

I'm currently using version 2.5.2.
May I ask which dataset you're using?

-------------------------------------------------------------------------------------------------------------------------------
> I have also conducted tests, and the accuracy on the ABIDE dataset is around 50%. This situation does exist in my case.

I haven't worked with the ABIDE dataset before, and as far as I know, the BrainGNN paper did not report results using ABIDE either.
